modules/consentManagement.js

```diff
@@ -84,6 +88,10 @@ function lookupIabConsent({onSuccess, onError}) {
         processCmpData(tcfData, {onSuccess, onError});
       } else {
         provisionalConsent = tcfData;
+        if (!iabCmpHasLoaded && !isNaN(actionTimeout) && shouldWaitForActionTimeout && timer != null) {
+          loadConsentData(loadIfMissingCb, true);
```

Comment:
This adds a lot of complexity that I think is unnecessary. `loadConsentData` now needs to do two completely different things based on the argument; I think it makes more sense to not mix them.

Why not just start a timeout here, as you do in line 255? To me it looks like it would get rid of all the other changes you have here.